src/types/entities/schelling.ts

```diff
@@ -37,72 +37,142 @@ export class SchellingGame {
 	// All players have a unique overlay address
 	private players: BTree_<string, Player>
 	private rounds: BTree_<number, Round>
+	private currentRoundNo = 0
+	private lastBlock: BlockDetails
 
 	private myOverlays: string[] = []
 	private myAccounts: string[] = []
 
+	private static _showLogs = false
+
 	/**
 	 * Make the constructor private so that it can't be called.
 	 * @private
 	 */
 	private constructor() {
 		this.players = new BTree()
 		this.rounds = new BTree()
+		this.currentRoundNo = 0 // We haven't started the game yet
+		this.lastBlock = { blockNo: 0, blockTimestamp: 0 }
+	}
+
+	public highlightOverlay(overlay: string) {
+		if (!this.myOverlays.includes(overlay)) {
+			this.myOverlays[this.myOverlays.length] = overlay
+			this.players.set(
+				overlay,
+				new Player(overlay, undefined, undefined, this.size)
+			)
+		}
 	}
 
 	// --- game logic ---
 
-	// --- commit
-	public commit(overlay: string, block: BlockDetails) {
+	public newBlock(block: BlockDetails): string {
 		const roundNo = SchellingGame.roundFromBlockNo(block.blockNo)
+		if (roundNo != this.currentRoundNo) {
+			if (this.currentRoundNo != 0) {
+				const round = this.rounds.get(this.currentRoundNo)
+				if (round && !round.claim) {
+					round.lastBlock = this.lastBlock
+					round.unclaimed = true
+					round.render()
+				}
+			}
+			this.currentRoundNo = roundNo
+		}
+		this.lastBlock = block
+		const blocksPerRound = config.blocksPerRound // TODO use configured blocksPerRound
```

Comment:
Looks like it actually has completed the TODO, but the following code is particularly inelegant, IMHO.